### PR TITLE
MV3: fix injection of applyLavamoat variable in service worker

### LINKED ISSUE #14787: [Bug]: @metamask/smart-transactions-controller>@metamask/controllers
### Describe the bug

app-init.js:8 Error: LavaMoat - required package not in allowlist: package "@metamask/smart-transactions-controller>@metamask/controllers" requested "@metamask/smart-transactions-controller>@metamask/controllers>@metamask/metamask-eth-abis" as "@metamask/metamask-eth-abis"

### Steps to reproduce

Error after mv3 branch

### Error messages or log output

_No response_

### Version

mv3

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jpuri**
Fixes: https://github.com/MetaMask/metamask-extension/issues/14787
Dynamically injecting `applyLavaMoat` in MV3 service worker.

### CODE REVIEW

File: app/scripts/app-init.js [hunk 1]
```diff
@@ -12,10 +12,8 @@ function tryImport(...fileNames) {
 
 function importAllScripts() {
   const startImportScriptsTime = Date.now();
-  // applyLavaMoat has been hard coded to "true" as
-  // tryImport('./runtime-cjs.js') is giving issue with XMLHttpRequest object which is not avaialble to service worker.
-  // we need to dynamically inject values of applyLavaMoat once this is fixed.
-  const applyLavaMoat = true;
+  // value of applyLavaMoat below is dynamically replaced at build time with actual value
+  const applyLavaMoat = false;
```
Comment (kumavis) [inline app/scripts/app-init.js:16]: default to true (always default to safe)

File: app/scripts/app-init.js [hunk 2]
```diff
@@ -25,14 +23,16 @@ function importAllScripts() {
...
     tryImport('./lockdown-install.js');
-    tryImport('./lockdown-more.js');
     tryImport('./lockdown-run.js');
+    tryImport('./lockdown-more.js');
     tryImport('./runtime-cjs.js');
   }
 
   const fileList = [
     // The list of files is injected at build time by replacing comment below with comma separated strings of file names
+    // https://github.com/MetaMask/metamask-extension/blob/develop/development/build/scripts.js#L406
```
Comment (kumavis) [inline app/scripts/app-init.js:35]: if you're going to put a reference on github, press the y key so you can lock to a commit and the line number is fixed to what you want to show

File: app/scripts/init-globals.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/**
+ * This script add properties in globalThis and initialises them with undefined.
+ * This is workaround needed to avoid error in runtime-cjs.js which is thrown when loading
```
Comment (kumavis) [inline app/scripts/init-globals.js:3]: error is not in runtime-cjs, error is in dependencies expecting to be run in a browser window and not a worker

**Review by kumavis — CHANGES_REQUESTED** [review]

File: app/scripts/app-init.js (hunk 2 above)
Comment (jpuri) [reply app/scripts/app-init.js:35]: so useful I did not knew this before

File: app/scripts/init-globals.js (hunk 3 above)
Comment (jpuri) [reply app/scripts/init-globals.js]: This is changed now.

File: app/scripts/app-init.js (hunk 1 above)
Comment (jpuri) [reply app/scripts/app-init.js]: updated now

**Review by kumavis — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM, but did you link the right issue? I'm not quite sure how it relates?

**Comment by jpuri** [comment]
> LGTM, but did you link the right issue? I'm not quite sure how it relates?

Hey @kumavis , @adonesky1 : issue linked is caused by `applyLavamoat` being true for `MV3` it will be fixed after this merge as `applyLavamoat` will be false in development mode.

**Pull request merged**